.github/workflows/tests.yml

```diff
@@ -0,0 +1,153 @@
+name: Simulator tests
+
+on:
+  push: {}
+  release:
+    types: [ created ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Build checksum file
+      run: |
+        declare -a FILES=(
+          "script/linting/requirements.txt"
+          "script/typing/requirements.txt"
+        )
+        for f in ${FILES[@]}
+        do
+          echo "# $f" >> all-requirements.txt
+          cat $f >> all-requirements.txt
+        done
+    - name: Cache dependencies
+      id: cache-dep
+      uses: actions/cache@v2
+      with:
+        path: venv
+        key: deps-py${{ matrix.python-version }}-${{ hashfiles('all-requirements.txt') }}
+    - name: Install Dependencies
+      if: steps.cache-dep.outputs.cache-hit != 'true'
+      run: |
+        python3 -m venv venv
+        venv/bin/pip install -U pip
+        venv/bin/pip install -r all-requirements.txt
+    - name: Test with unittest
+      run: |
+        source venv/bin/activate
+        ./script/testing/test
+
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Build checksum file
+      run: |
+        declare -a FILES=(
+          "script/linting/requirements.txt"
+          "script/typing/requirements.txt"
+        )
+        for f in ${FILES[@]}
+        do
+          echo "# $f" >> all-requirements.txt
+          cat $f >> all-requirements.txt
+        done
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      id: cache-dep
+      with:
+        path: venv
+        key: deps-py${{ matrix.python-version }}-${{ hashfiles('all-requirements.txt') }}
+    - name: Install Dependencies
+      if: steps.cache-dep.outputs.cache-hit != 'true'
+      run: |
+        python3 -m venv venv
+        venv/bin/pip install -U pip
+        venv/bin/pip install -r all-requirements.txt
+    - name: Lint with flake8
+      run: |
+        source venv/bin/activate
+
+        # Ignore TODOs on CI, for now, even though we do want them
+        # highlighted in development.
+        ./script/linting/lint --extend-ignore=T000
+
+  typecheck:
```

Comment:
Given the steps between linting and typechecking are nearly identical, probably makes sense to just combine the steps?

Reply:
We could also merge test and essentially run the check script against all python versions